spec/wibox/widget/textbox_spec.lua

```diff
@@ -113,6 +124,29 @@ describe("wibox.widget.textbox", function()
             assert.is.equal(2, layout_changed)
         end)
     end)
+
+    describe("auxiliary", function()
+
+        it("can compute text geometry w/default font", function()
+            local text = "<b><i>test</i></b>"
+            local s = 1
+            local pango_geometry = textbox.get_markup_geometry(text, s)
+            local actual_textbox_width, actual_textbox_height = textbox(text):get_preferred_size(s)
+            assert.is.equal(pango_geometry.width, actual_textbox_width)
+            assert.is.equal(pango_geometry.height, actual_textbox_height)
+        end)
+
+        it("can compute text geometry w/hardcoded font", function()
+            local text = "<span font='Monospace 16'><b><i>test</i></b></span>"
+            local s = 1
+            local pango_geometry = textbox.get_markup_geometry(text, s)
+            local actual_textbox_width, actual_textbox_height = textbox(text):get_preferred_size(s)
+            assert.is.equal(pango_geometry.width, actual_textbox_width)
+            assert.is.equal(pango_geometry.height, actual_textbox_height)
+        end)
```

Comment:
The control value is computed in the same way that `get_markup_geometry` uses internally. This means that this test doesn't actually test whether the return value of `get_markup_geometry` is correct. It merely ensures that `get_markup_geometry` doesn't alter the result of `get_preferred_size` before returning it, while that result may still be incorrect.
It also means that this test is tied to implementation details of the function being tested, which should not be the case for a unit test.

You should get rid of the "default font" test variant completely and change the "hardcoded font" variant to
- use an actual "hardcoded font", e.g. `Terminus 8` or `Dejavu 10`. `Monospace` isn't really "hardcoded", since that still depends on how the system is configured that runs the test.
- determine what the result of the tested function should be (i.e. run it once with the test's inputs) and provide that as fixed numeric literals in the `assert`. Also, the first parameter is the `expected` one
- change the `assert`s to this schema: `assert.is.same({ width = 10, height = 10 }, pango_geometry)`. This way the test also ensures that no unexpected stuff may appear in the result table.

Reply:
the bikeshed discussion for implementing this trivial height computation might continue for 2 more years, as every next reviewer have the opposite opinion to previous one, so i'll first wait for 2 same opinions on this, before changing smth further